prov/gni/src/gnix_hashtable.c

```diff
@@ -715,9 +724,6 @@ int _gnix_ht_remove(gnix_hashtable_t *ht, gnix_ht_key_t key)
 
 void *_gnix_ht_lookup(gnix_hashtable_t *ht, gnix_ht_key_t key)
 {
-	if (ht->ht_state != GNIX_HT_STATE_READY)
-		return NULL;
-
```

Comment:
I think you're saying is that this is *not* expected behavior from this interface.  Then I'm fine with no assert.